src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb

```diff
@@ -376,71 +373,68 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
     Partial Class BoundForToStatement
         Implements IForLoopStatement
 
-        Private Shared ReadOnly s_loopBottomMappings As New System.Runtime.CompilerServices.ConditionalWeakTable(Of BoundForToStatement, Object)
+        Private _lazyLoopBottom As ImmutableArray(Of IStatement)
```

Comment:
In the interest of minimizing the sizes of bound nodes for which these are never accessed, it's probably better to introduce only one field to hold all IOperation-related information.